cmd/clusterctl/pkg/client/cluster/proxy.go

```diff
@@ -70,12 +69,7 @@ func (k *proxy) NewClient() (client.Client, error) {
 		return nil, errors.Wrap(err, "failed to create controller-runtime client")
 	}
 
-	mapper, err := apiutil.NewDynamicRESTMapper(config, apiutil.WithLazyDiscovery)
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create the controller-runtime DynamicRESTMapper")
-	}
-
-	c, err := client.New(config, client.Options{Scheme: Scheme, Mapper: mapper})
```

Comment:
@fabriziopandini is this proxy client a long lived client within `clusterctl` that would require the client to be setup and configured prior to the remote server listening? If not, then I think we can get away without lazy discovery and use the default controller-runtime client mapper.

Reply:
This client is used in many places, usually short-lived.
However, there are places like e.g. when installing cert-manager, where we create CRDs and immediately after we check for the CRDs to be available, and without the NewDynamicRESTMapper the client was not able to detect the new type (no matter if creating a new client or not)

Reply:
@fabriziopandini with the version of controller-runtime that we updated to use of the Dynamic REST Mapper is now the default. The behavior change introduced here is removal of the lazy discovery (which should only be needed if we absolutely need to be able to create a client before the server is available).